### PR TITLE
Removed RxJava from Gradle plugin, using Kotlin stdlib instead

### LINKED ISSUE #258: Remove RxJava from plugin. Use Kotlin's stdlib.

### PR DESCRIPTION
**Pull request by grodin**
Closes #258

### CODE REVIEW

**Comment by grodin** [comment]
No problem. Very happy to help!

File: buildSrc/src/main/kotlin/com/jakewharton/rxbinding/project/ValidateBindingsTask.kt [hunk 1]
```diff
@@ -95,14 +94,11 @@ open class ValidateBindingsTask : SourceTask() {
...
-            .filter { it.type is ReferenceType }
-            .map { it.id.name },
-        Observable.from(statements),
-        { s, stmt -> Pair(s, stmt.toString()) }
-    )
-        .subscribe({
+    parameters
+        .filter { it.type is ReferenceType }
+        .map { it.id.name }
+        .zip(statements, { param, stmt -> Pair(param, stmt) })
```
Comment (JakeWharton) [inline buildSrc/src/main/kotlin/com/jakewharton/rxbinding/project/ValidateBindingsTask.kt]: It looks like you lost a `.toString()` on `stmt` here which is causing the failure.

Comment (JakeWharton) [inline buildSrc/src/main/kotlin/com/jakewharton/rxbinding/project/ValidateBindingsTask.kt]: You could also put it in the `forEach` below if you want.

Comment (JakeWharton) [inline buildSrc/src/main/kotlin/com/jakewharton/rxbinding/project/ValidateBindingsTask.kt]: Woah I was able to do this myself using GitHub's web editor in your branch. Cool!

**Comment by grodin** [comment]
Would you be interested if I added some unit tests to the plugin? 

I just checked my branch again, and it all worked locally: I ran `generateKotlin` which built fine, and `git status --porcelain` didn't show anything changed. So it seems it would probably be useful to have some local tests to check the logic in the plugin. 

It would be another PR obviously, but it seemed like an appropriate place to ask.

**Comment by JakeWharton** [comment]
Yeah you change works for generating Kotlin but I forgot we also add a validation task to the normal build. If you had run the `build` task as well to actually compile the sources it would have failed locally.

I'm not opposed to tests, but I just worry they'd be redundant because we exercise all functionality of the plugin in the normal build steps already.

**Comment by grodin** [comment]
Maybe just a README in the `buildSrc` project then, with an explanation how to check it works? It should be documented somewhere, just so you don't have to remember!

**Comment by JakeWharton** [comment]
Thanks!

**Pull request merged**